File: Solutions/1351. Count Negative Numbers in a Sorted Matrix/1351. Count Negative Numbers in a Sorted Matrix.cpp
class Solution {
public:
    int countNegatives(vector<vector<int>>& grid) {
        int res=0,m=grid.size(),n=grid[0].size();
        int r=m-1,c=0;
        while(r>=0 && c<n){
            if(grid[r][c]<0){
                res+=n-c;
                --r;
            }else
                ++c;
        }
        return res;
    }
};

File: Solutions/1351. Count Negative Numbers in a Sorted Matrix/1351. Count Negative Numbers in a Sorted Matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        res,m,n=0,len(grid),len(grid[0])
        r,c=m-1,0
        while r>=0 and c<n:
            if grid[r][c]<0:
                res+=n-c
                r-=1
            else:
                c+=1
        return res